### PR TITLE
Hide sprites via context menu addon

### LINKED ISSUE #7077: new addon: show/hide sprites via context menu
### Why this would be helpful

Moving from one sprite to another just to hide something can get annoying on large projects and doing some workaround such as via broadcasts will just take up memory and that's still not really a full solution

### How this addon works

adds an extra button to context menus to show or hide the sprite
either just include both or dynamically change to the opposite of the sprite's current state (shown/hidden)

### Possible alternatives

A new tab in sprites just for hiding/showing with all sprites listed (though this would be probably too big)
no idea for alternatives

One possible setting could be including both show and hide instead of replacing the option like with `block-switching`'s "show option to change block to self".

### Additional context

<sub>I used turbowarp and the scratch editor for so long and this has always been bugging me with the lag of going to another sprite and then switching back
no other context</sub>

### PR DESCRIPTION
**Pull request by mybearworld**
Resolves #7077

### Changes

Adds a new addon that adds a context menu entry that hides/shows a sprite.

A screenshot of the context menu with the option "hide".

### Reason for changes

Convenience, OneShot-Niko accidentally wanting to make duplicates 3 times

### Tests

Tested in Edge.

### CODE REVIEW

**Comment by OneShot-Niko** [comment]
Could this be enabled by default?

> OneShot-Niko accidentally wanting to make duplicates 3 times

in my defense my memory is absolutely horrible, atleast I managed to remember each time lol

File: addons/hide-show-context-menu/addon.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "Hide sprites via context menu",
```
Comment (Samq64) [inline addons/hide-show-context-menu/addon.json:2]:
```suggestion
  "name": "Hide sprites from context menu",
```

File: addons/hide-show-context-menu/userscript.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  const stateFromBool = (bool) => (bool ? "show" : "hide");
+  const sharedOptions = {
+    types: ["sprite"],
+    position: "assetContextMenuAfterDelete",
+    order: 9, // before folders - it's unpredictable how many slots folders takes up
+    border: true,
+  };
+
+  [true, false].forEach((visibility) => {
```
Comment (Samq64) [inline addons/hide-show-context-menu/userscript.js:13]: This works but it's confusing.

File: addons/hide-show-context-menu/userscript.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    addon.tab.createEditorContextMenu(
+      (ctx) => {
+        getSprite(ctx.itemId).setVisible(visibility);
+      },
+      {
+        ...sharedOptions,
+        className: `sa-hide-show-context-menu-${stateFromBool(visibility)}`,
```
Comment (Samq64) [inline addons/hide-show-context-menu/userscript.js:20]: This addon doesn't have a userstyle so I don't think a class is necessary.

**Review by Samq64 — APPROVED** [review]

**Comment by Samq64** [comment]
I didn't mean to approve but the addon does work as expected.

File: addons/hide-show-context-menu/addon.json (hunk 1 above)
Comment (mybearworld) [reply addons/hide-show-context-menu/addon.json:2]: This sounds to me like regularly, the sprites would appear in a context menu, and this addon hides them from it.

File: addons/hide-show-context-menu/userscript.js (hunk 2 above)
Comment (mybearworld) [reply addons/hide-show-context-menu/userscript.js:13]: The alternative would be code duplicating, wouldn't it?

File: addons/hide-show-context-menu/userscript.js (hunk 3 above)
Comment (mybearworld) [reply addons/hide-show-context-menu/userscript.js:20]: It's required by the function according to the docs.

File: addons/hide-show-context-menu/addon.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    }
+  ],
+  "credits": [
+    {
+      "name": "mybearworld",
+      "link": "https://scratch.mit.edu/users/mybearworld"
+    }
+  ],
+  "dynamicEnable": true,
+  "dynamicDisable": true
```
Comment (Samq64) [inline addons/hide-show-context-menu/addon.json:19]: Dynamic disable isn't handled right now.

File: addons/hide-show-context-menu/addon.json (hunk 1 above)
Comment (Samq64) [reply addons/hide-show-context-menu/addon.json:2]:
```suggestion
  "name": "Hide sprites with context menu",
```

File: addons/hide-show-context-menu/addon.json (hunk 4 above)
Comment (mybearworld) [reply addons/hide-show-context-menu/addon.json:19]: The function seems to handle that too.

**Pull request closed without merging**